boards/xtensa/heltec_wifi_lora32/heltec_wifi_lora32_defconfig

```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_XTENSA_RESET_VECTOR=n
+
+CONFIG_BOARD_HELTEC_WIFI_LORA32=y
+CONFIG_SOC_ESP32=y
```

Comment:
Default configs also seem to be same for all ESP32 boards. We can move it to a common file?

Reply:
It's a good idea. I looked at other boards but I couldn't figure out where to put a common file and make both boards/xtensa/esp32 and boards/xtensa/heltec_wifi_lora32_v2 use it. Do you have a recommendation?

Reply:
It looks like @sylvioalves is working on some thing like this. Should we keep this PR as is and move the common config on another PR? (The same for CMake files)

Reply:
Let's keep as is. Once we refactor board folder structure we re-work this properly.